Sources/OpenAI/OpenAI.swift

```diff
@@ -46,7 +46,7 @@ final public class OpenAI: OpenAIProtocol {
         self.init(configuration: configuration, session: URLSession.shared)
     }
     
-    init(configuration: Configuration, session: URLSessionProtocol = URLSession.shared) {
+    public init(configuration: Configuration, session: URLSessionProtocol = URLSession.shared) {
```

Comment:
The protocols should be public just like `OpenAIProtocol`; this way it adheres best to the dependency inversion principle: interfaces are declared on the callee, and the caller (dependent) decides how to supply the value. The caller can choose to supply with a concrete implementation we provide (`OpenAI`), or to provide a mocked type.

Reply:
Thanks for the clarification, @DJBen 👍 
I agree with the concept, but I don't see any applicable approach. Having `OpenAIProtocol`  instead of concrete `OpenAI` gives the ability to mock its implementation so the code that is dependent on it is available for unit tests, which is great.  
Same with `URLSessionProtocol`, but in this case dependent code is `OpenAI` class, which is not owned by its users, and should be unit tested internally. 
Like we use Foundation API, for example [FileManager](https://developer.apple.com/documentation/foundation/filemanager/3025773-init)
```swift
convenience init(authorization: NSWorkspace.Authorization)
```
Since we don't own FileManger's implementation, we don't have to unit-test it, but if our code is dependent on it, we can handle it by creating `FileMangerType` to mock it's implementation and unit-test our code. 
I'm ok to make the protocols public, but don't see any practical reason for now. 

What do you think?

Reply:
There's no practical reason to do so, yet in practice I see no harm doing it. If multiple modules have the same mock (which is bound to happen in a large project, we can reference them separately through module prefixes. Although I am okay with the approach to unblock #50 asap.

```swift
public init(configuration: Configuration, session: URLSession)
```

Reply:
Updated